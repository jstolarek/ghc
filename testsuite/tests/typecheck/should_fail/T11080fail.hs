-- JSTOLAREK: some language extension here
{-# LANGUAGE GADTSyntax #-}
module T11080fail where

data kind Foo = Bar

-- this should be rejected
foo :: Foo
foo = Bar

data kind Foo' where
    Bar'  :: Foo'

bar :: Foo'
bar = Bar'
