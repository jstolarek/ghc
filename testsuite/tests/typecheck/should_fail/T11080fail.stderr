
T11080fail.hs:9:7:
     Bar is used to construct an expression but it is allowed only in types
     In the expression: Bar
      In an equation for ‘foo’: foo = Bar

T11080fail.hs:15:7:
     Bar' is used to construct an expression but it is allowed only in types
     In the expression: Bar'
      In an equation for ‘bar’: bar = Bar'
